### ISSUE TITLE
Document Function and Classes

### ISSUE THREAD

**Issue by tomschr** [openSUSE/docmanager#7]
Functions and classes needs to be documented: their purpose, arguments, and argument types. This makes maintenance a lot easier.

There are several ways to do this:
- [Sphinx style](http://sphinx-doc.org/domains.html#info-field-lists)
- [Google style](http://google-styleguide.googlecode.com/svn/trunk/pyguide.html#Comments)
- [NumPy style](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt)

IMHO, Sphinx style would be appropriate. Here is a short example in Sphinx style:

``` python
def add(a, b):
    """Adds two numbers

    :param int a: first argument
    :param int b: second argument
    :return:      sum of a and b
    :rtype:       int
    """
    return a+b
```

**Comment by tomschr** [comment]
@rsalevsky I've documented almost all function and classes in Sphinx style. However, I was unsure of some functions, their datatypes and purpose. 

Could you review and amend the docstrings, please? Look for the "TODO" keyword to simplify this task.

Thanks! :-)

**Comment by rsalevsky** [comment]
@tomschr LGTM so far, makes the code much more easier to read

**Comment by tomschr** [comment]
@rsalevsky Thanks. This is now merged from 3830fecd0c4e70ae6b6308b8e511556343d6c9e8 into develop.

However, there are still some functions which were unclear to me. These are still marked as "TODO". ;-) My idea was that _you_ go through this list and add the missing parts. It's only 10 spots, easy to fix.

**Comment by tomschr** [comment]
@rsalevsky Your TODOs isn't needed anymore. :smile: so closing
